stdlib/LinearAlgebra/test/matmul.jl

```diff
@@ -424,7 +425,7 @@ end
     @test  transpose(Xv2)*Xv2     ≈ XtX
     @test (transpose(Xv3)*Xv3)[1] ≈ XtX
     @test  Xv1'*Xv1     ≈ XcX
-    @test  Xv2'*Xv2     ≈ XcX
+    @test_throws MethodError Xv2'*Xv2 ≈ XcX
```

Comment:
Why the transformation to a MethodError?

Reply:
This is mainly due to the change of the behaviour of `dot` acting recursively. `Xv2` is a vector of transposed vectors and `Xv2'*Xv2` is defined as `dot(Xv2, Xv2)`. Thus, the left-hand sides gives a scalar while the right-hand side gives a 3x3 matrix. Without recursive definition, the left-hand side has been `sum(Xv2[i]' * Xv2[i] for i in eachindex(Xv2))` (`== XcX`).

Reply:
The new behaviour seems to be consistent with most of the discussion here and in #25565: `Xv2` is a vector and `Xv2' * Xv2` should be the dot/inner/scalar product of `Xv2` with itself, i.e. a scalar.

Reply:
Probably `@test  Xv2'*Xv2 ≈ norm(Xv2)` in that case?

Reply:
Thanks, I've added such a test.